cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/AbstractJavaTemplateTest.java

```diff
@@ -0,0 +1,117 @@
+package com.devonfw.cobigen.templates.devon4j.test.templates;
+
+import com.devonfw.cobigen.api.extension.TextTemplate;
+import com.devonfw.cobigen.javaplugin.inputreader.JavaInputReader;
+import com.devonfw.cobigen.tempeng.freemarker.FreeMarkerTemplateEngine;
+import org.junit.Before;
+
+import java.io.StringWriter;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+
+public abstract class AbstractJavaTemplateTest {
+  /**
+   * Model used by freemarker
+    */
+  public Map<String, Object> model;
+  public FreeMarkerTemplateEngine engine;
+  /**
+   * Implementation of TextTemplate for template processing
+   */
+  public TextTemplate template;
+
+  /**
+   * Creates an anonymous TextTemplate object that works on the given template files
+   * 
+   * @param relativePath Relative Path to the template from the /templates folder, most likely coincides with template
+   *        filename
+   * @param relativeAbsolutePath Relative path from source root to the template
+   */
+  public TextTemplate createTemplate(String relativePath, Path relativeAbsolutePath) {
+
+    return new TextTemplate() {
+      @Override
+      public String getRelativeTemplatePath() {
+
+        return relativePath;
+      }
+
+      @Override
+      public Path getAbsoluteTemplatePath() {
+
+        return relativeAbsolutePath;
+      }
+    };
+  }
+
+  /**
+   * Creates a template engine for the given template path
+   * 
+   * @param templateFolderPath Relative path to template folder from project source root
+   */
+  public FreeMarkerTemplateEngine createEngine(String templateFolderPath) {
```

Comment:
Javadoc for return missing.

Reply:
Is return type and method name not self-explanatory?